src/compiler/program.ts

```diff
@@ -2416,10 +2416,26 @@ namespace ts {
             return options.jsx ? undefined : Diagnostics.Module_0_was_resolved_to_1_but_jsx_is_not_set;
         }
         function needAllowJs() {
-            return options.allowJs || !getStrictOptionValue(options, "noImplicitAny") ? undefined : Diagnostics.Could_not_find_a_declaration_file_for_module_0_1_implicitly_has_an_any_type;
+            return willAllowImplicitAnyJsModule(options) ? undefined : Diagnostics.Could_not_find_a_declaration_file_for_module_0_1_implicitly_has_an_any_type;
         }
     }
 
+    export function willAllowImplicitAnyJsModule(options: CompilerOptions): boolean {

```

Comment:
`/* @internal */`

Reply:
consider moving this to untilities next to `getAllowSyntheticDefaultImports` and `getStricitOptionValue`